metricbeat/tests/system/test_processors.py

```diff
@@ -39,8 +39,10 @@ def test_drop_fields(self):
         ]), evt.keys())
         network = evt["system"]["network"]
         print(list(network.keys()))
-        self.assertCountEqual(self.de_dot(["name", "out", "in"]), network.keys())
+        self.assertCountEqual(self.de_dot(
+            ["name", "out", "in"]), network.keys())
 
+    @unittest.skipUnless(re.match("(?i)linux|darwin|freebsd", sys.platform), "os")
```

Comment:
Having to disable tests with a big refactor like this for Windows concerns me a little bit if we don't know what the root cause is. Can we prove that the tests are equally flaky on main?

Reply:
Yah, I'm super baffled by this, and I don't have enough experience with the testing setup or windows. I can run them on my windows boxes, and the tests will pass, but the entire suite of unit tests, not just the ones impacted by the refactor, are super flaky on any windows machine I test. The tests I commented out on windows are flaky as there's a small chance that the parallel unit tests will return "wrong" events that don't have the full processor data.

I assume I can fix this by refactoring the python tests to add a few filter functions, currently not sure if that's worth it, considering these tests are 6 years old.

Reply:
There is nothing about Windows in particular that should cause flakiness. If you are testing on older hardware sometimes the slightly slower execution speed can reveal timing issues that result in flakiness.

I wouldn't focus on the age of the tests but whether they are actually providing valuable test coverage. If they are it is worth trying to improve them (or just replace them with something better like an equivalent Go test). If they aren't testing anything useful we could remove them.

Reply:
Yah, the flakiness is really odd, considering it's running on a fairly nice windows PC, but I'm still seeing inconsistent timeouts. I think it's worth trying to improve the tests, but they need enough cleanup/additions that I'm wondering if I should do it in a separate PR, since they need a good bit of cleanup.